MDK Showcase App - MDK consuming SAP AI Core services

gqswang opened the pull request:
This project serves as a Demo App for consuming SAP AI Core services and SAP HANA Vector DB service with the SAP Mobile Development Kit (MDK). It demonstrates the following use cases:

1. Extracting meter readings from images.
2. Detecting anomalies using images.
3. Generating operations based on text-described anomalies.
4. Auto-filling forms using speech-to-text conversion.

qswangsap committed:
Initial version

qswangsap committed:
Migrate to MDK Action.Type.AICore.Chat.Completions

gqswang committed:
Merge branch 'main' into Migrate-to-MDK-Action.Type.AICore.Chat.Completions

qswangsap committed:
Migrate to MDK Action.Type.AICore.Chat.Completions

gqswang committed:
Merge pull request #1 from gqswang/Migrate-to-MDK-Action.Type.AICore.Chat.Completions

Migrate to mdk action.type.ai core.chat.completions

qswangsap committed:
Update README for MDK web client

qswangsap committed:
Migrate metadata with the latest schema updates from MDK 24.11.x

gqswang committed:
Merge pull request #1 from qswangsap/main

add back MDKGenAI.mdkproject

jitendrakansal commented on Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md:
```diff
@@ -0,0 +1,108 @@
+# MDK Generative AI 
```
```suggestion
# AI-Enabled Field Service App
```

jitendrakansal commented on Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md:
```diff
@@ -0,0 +1,108 @@
+# MDK Generative AI 
+
+This project serves as a Demo App for consuming SAP AI Core services and SAP HANA Vector DB service with the SAP Mobile Development Kit (MDK). It demonstrates the following use cases:
```
```suggestion
This project serves as a demo app for consuming SAP AI Core and SAP HANA Vector DB services using the SAP Mobile Development Kit (MDK). It demonstrates the following use cases:
```

jitendrakansal commented on Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md:
```diff
@@ -0,0 +1,108 @@
+
+This project serves as a Demo App for consuming SAP AI Core services and SAP HANA Vector DB service with the SAP Mobile Development Kit (MDK). It demonstrates the following use cases:
+
+1. Extracting meter readings from images.
```
```suggestion
1. Extracting meter readings from images
```

jitendrakansal commented on Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md:
```diff
@@ -0,0 +1,108 @@
+This project serves as a Demo App for consuming SAP AI Core services and SAP HANA Vector DB service with the SAP Mobile Development Kit (MDK). It demonstrates the following use cases:
+
+1. Extracting meter readings from images.
+2. Detecting anomalies using images.
```
```suggestion
2. Detecting anomalies from images
```

jitendrakansal commented on Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md:
```diff
@@ -0,0 +1,108 @@
+
+1. Extracting meter readings from images.
+2. Detecting anomalies using images.
+3. Generating operations based on text-described anomalies.
```
```suggestion
3. Generating operations based on text-described anomalies
```

jitendrakansal commented on Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md:
```diff
@@ -0,0 +1,108 @@
+1. Extracting meter readings from images.
+2. Detecting anomalies using images.
+3. Generating operations based on text-described anomalies.
+4. Auto-filling forms using speech-to-text conversion.
```
```suggestion
4. Auto-filling forms using speech-to-text conversion
```

jitendrakansal commented on Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md:
```diff
@@ -0,0 +1,108 @@
+
+## Getting Started
+### Prerequisites
+1. Create an instance of SAP AI Core and then create a Service Key.
```
```suggestion
1. Create an instance of SAP AI Core and then create a service key.
```

jitendrakansal commented on Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md:
```diff
@@ -0,0 +1,108 @@
+### Prerequisites
+1. Create an instance of SAP AI Core and then create a Service Key.
+
+* refer to https://developers.sap.com/tutorials/ai-core-launchpad-provisioning.html and https://developers.sap.com/tutorials/ai-core-generative-ai.html for details on how to setup SAP AI Core and SAP AI Launchpad
```
```suggestion
* Refer to the following tutorials for details on how to set up SAP AI Core and SAP AI Launchpad:
- https://developers.sap.com/tutorials/ai-core-launchpad-provisioning.html
- https://developers.sap.com/tutorials/ai-core-generative-ai.html

```

jitendrakansal commented on Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md:
```diff
@@ -0,0 +1,108 @@
+    deleteEmbeddings(): returns String
+    getRagResponse(value: String): returns String
+
+For this demo, the embeddings are configured in PopulateEmbeddings.js
```
```suggestion
For this demo, the embeddings are configured in PopulateEmbeddings.js.
```

jitendrakansal commented on Showcase_Apps/AICoreApp/README.md:
```diff
@@ -0,0 +1,108 @@
+
+### Demo
+
+##### Blogs
```
formatting of blog links doesnt appear nicely.. pls. look and add put them in some bullet list.

jitendrakansal commented on Showcase_Apps/AICoreApp/README.md:
````diff
@@ -0,0 +1,108 @@
+
+
+```bash
+Name: AICoreAPI
````
Is this a BTP destination? Why do we need it?

gqswang replied:
yes, it's BTP destination, in Mobile Service, we will use this BTP destination to configure mobile connectivity for AI Core Service

jitendrakansal replied:
Got it. But then we should recommend creating a new mobile destination with all details directly in mobile services instead of proxying it via BTP destination.

gqswang committed:
Update Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md

Co-authored-by: JITENDRA KANSAL <jitendra.kansal@sap.com>

gqswang committed:
Update Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md

Co-authored-by: JITENDRA KANSAL <jitendra.kansal@sap.com>

gqswang committed:
Update Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md

Co-authored-by: JITENDRA KANSAL <jitendra.kansal@sap.com>

gqswang committed:
Update Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md

Co-authored-by: JITENDRA KANSAL <jitendra.kansal@sap.com>

gqswang committed:
Update Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md

Co-authored-by: JITENDRA KANSAL <jitendra.kansal@sap.com>

gqswang committed:
Update Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md

Co-authored-by: JITENDRA KANSAL <jitendra.kansal@sap.com>

gqswang committed:
Update Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md

Co-authored-by: JITENDRA KANSAL <jitendra.kansal@sap.com>

gqswang committed:
Update Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md

Co-authored-by: JITENDRA KANSAL <jitendra.kansal@sap.com>

gqswang committed:
Update Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md

Co-authored-by: JITENDRA KANSAL <jitendra.kansal@sap.com>

gqswang committed:
Update demo blog links in README.md

jitendrakansal commented on Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md:
```diff
@@ -0,0 +1,111 @@
+### Prerequisites
+1. Create an instance of SAP AI Core and then create a service key.
+
+* Refer to the following tutorials for details on how to set up SAP AI Core and SAP AI Launchpad:
```
```suggestion
    > Refer to the following tutorials for details on how to set up [SAP AI Core](https://developers.sap.com/tutorials/ai-core-generative-ai.html) and [SAP AI Launchpad](https://developers.sap.com/tutorials/ai-core-launchpad-provisioning.html).
```

gqswang committed:
Update Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md

Co-authored-by: JITENDRA KANSAL <jitendra.kansal@sap.com>

jitendrakansal commented on Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md:
```diff
@@ -11,7 +11,7 @@ This project serves as a demo app for consuming SAP AI Core and SAP HANA Vector
 
-* Refer to the following tutorials for details on how to set up SAP AI Core and SAP AI Launchpad:
+    > Refer to the following tutorials for details on how to set up [SAP AI Core](https://developers.sap.com/tutorials/ai-core-generative-ai.html) and [SAP AI Launchpad](https://developers.sap.com/tutorials/ai-core-launchpad-provisioning.html).
 - https://developers.sap.com/tutorials/ai-core-launchpad-provisioning.html
```
we dont need these two links now as these are already there as hyperlinked.

gqswang replied:
removed in [c3acd3f](https://github.com/SAP-samples/cloud-mdk-samples/pull/58/commits/c3acd3f454af4c7dc67d18522961b286a5c3238f)

jitendrakansal commented on Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md:
````diff
@@ -0,0 +1,111 @@
+URL.headers.Content-Type: application/json
+HTML5.DynamicDestination: true
+```
+4. Create a similar destination in Mobile Services for the deployed service using the Service Key of SAP AI Core for Mobile.
````
```suggestion
3. Create a similar destination in Mobile Services for the deployed service using the Service Key of SAP AI Core for Mobile.
```

jitendrakansal reviewed (changes requested):
added

gqswang committed:
Update Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md

Co-authored-by: JITENDRA KANSAL <jitendra.kansal@sap.com>

gqswang committed:
Clean up README by removing duplicate links

Removed duplicate tutorial links for SAP AI Core and Launchpad.

jitendrakansal reviewed (approved):
lgtm

jitendrakansal commented:
@billfroelich pls. review this PR

jitendrakansal committed:
test changes

jitendrakansal committed:
removed test updates

billfroelich committed:
Update README

Moved readme to top level of showcase directory and updated / added screenshots

jitendrakansal committed:
updated readme

jitendrakansal committed:
adjusted destination configuration part

jitendrakansal committed:
added screenshot

jitendrakansal committed:
Update README.md

jitendrakansal committed:
Merge branch 'main' into main
